src/components/Header/Header.jsx

```diff
@@ -1,29 +1,49 @@
-import { Link, useNavigate } from "react-router-dom";
+import { Link, useLocation, useNavigate } from "react-router-dom";
 import { useDispatch } from "react-redux";
 import useAuth from "hooks/useAuth";
+import useSearch from "hooks/useSearch";
 import { loggedOutUser } from "features/auth/authSlice";
 import LogoutIcon from "components/icons/LogoutIcon";
 import SearchIcon from "components/icons/SearchIcon";
-import "./Header.scss";
+import {
+  updateHomeQuery,
+  updateProfileQuery,
+} from "features/search/searchSlice";
+import "./header.scss";
 
 const Header = () => {
   const { authUser } = useAuth();
-
+  const { queryInHome, queryInProfile } = useSearch();
   const navigate = useNavigate();
   const dispatch = useDispatch();
+  const location = useLocation();
+  const profile = location.pathname === "/me";
 
   const handleLogout = () => {
     dispatch(loggedOutUser());
     navigate("/login");
   };
 
+  const handleSearch = (event) => {
+    if (profile) {
+      dispatch(updateProfileQuery(event.target.value));
+      return;
+    }
+    dispatch(updateHomeQuery(event.target.value));
+  };
+
   return (
     <nav className="navbar">
       <Link to="/" className="navbar__title">
         WellBlog
       </Link>
       <div className="navbar__search">
-        <input type="search" placeholder="Search" />
+        <input
```

Comment:
Shouldn't this be in a field? and can we use initialValue instead of value prop in that Field?

Reply:
@ray-wtag bhai, is it necessary to use any form package for a single input field?